FreeRTOS/FreeRTOS-Plus-TCP
# Discard UDPv6 with zero checksum.

Discard UDPv6 with zero checksum.

Description
-----------

Refer to [RFC2460 - section 8.1](https://www.rfc-editor.org/rfc/rfc2460#section-8.1), the UDPv6 packets containing a zero checksum must be discarded:
```
Unlike IPv4, when UDP packets are originated by an IPv6 node,
the UDP checksum is not optional.  That is, whenever
originating a UDP packet, an IPv6 node must compute a UDP
checksum over the packet and the pseudo-header, and, if that
computation yields a result of zero, it must be changed to hex
FFFF for placement in the UDP header.  IPv6 receivers must
discard UDP packets containing a zero checksum, and should log
the error.
```


Test Steps
-----------

Run protocol test case UDP/checksum/004 with traffic source udp.v6.

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
